packages/ibm-products/src/components/Datagrid/useNestedRows.js

```diff
@@ -64,11 +64,21 @@ const useNestedRows = (hooks) => {
         props,
         {
           style: {
-            marginRight: `${
+            marginRight:
+              isFirstCell &&
```

Comment:
Hey ,
Looks good and seems to be working.
Can you add some comments regarding the logic and figures used here for calculation?

Reply:
done some refactors for readability, and added comments